### ISSUE TITLE
CA1850 (ComputeHash) code fix removes namespace erroneously

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#5682]
### Analyzer

**Diagnostic ID**: [CA1850](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1850): 'Prefer HashData over ComputeHash`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0-preview1.21523.2

### Describe the bug

CA1850 removes the namespace from a fully-qualified name even if the required namespace isn't imported.

### Steps To Reproduce

```C#
internal sealed class Repro
{
    public static byte[] M(byte[] data)
    {
        using (var sha = System.Security.Cryptography.SHA256.Create())
        {
            return sha.ComputeHash(data);
        }
    }
}
```
Then run the code fix for CA1850.

### Expected behavior

```C#
internal sealed class Repro
{
    public static byte[] M(byte[] data)
    {
        return System.Security.Cryptography.SHA256.HashData(data);
    }
}
```

### Actual behavior

```C#
internal sealed class Repro
{
    public static byte[] M(byte[] data)
    {
        return SHA256.HashData(data);
    }
}
```
